src/theory/bv/bv_solver.h

```diff
@@ -0,0 +1,119 @@
+/*********************                                                        */
+/*! \file bv_solver.h
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Mathias Preiner
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2020 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief Bit-vector solver interface.
+ **
+ ** Describes the interface for the internal bit-vector solver of TheoryBV.
+ **/
+
+#include "cvc4_private.h"
+
+#ifndef CVC4__THEORY__BV__BV_SOLVER_H
+#define CVC4__THEORY__BV__BV_SOLVER_H
+
+#include "theory/bv/theory_bv.h"
+#include "theory/theory.h"
+
+namespace CVC4 {
+namespace theory {
+namespace bv {
+
+class BVSolver
+{
+ public:
+  BVSolver(TheoryBV& bv) : d_state(bv.d_state), d_inferManager(bv.d_infer){};
```

Comment:
I think we should just pass reference to a state and inference manager here.  The constructor for BVSolverLazy can take all 3 if it needs (or just TheoryBV from which these can be extracted), but we can avoid TheoryBV here.